File: 1567-maximum-number-of-vowels-in-a-substring-of-given-length/maximum-number-of-vowels-in-a-substring-of-given-length.cpp
class Solution {
public:
    int maxVowels(string s, int k) {
        unordered_set<char> v = {'a', 'e', 'i', 'o', 'u'};
        int i = 0, j = 0;
        int ans = 0, count = 0;
        while (j < s.length()) {
            char c = s[j];
            if (j - i + 1 < k) {
                if (v.count(c)) {
                    count += 1;
                }
            } else if (j-i+1 == k){
                if (v.count(c)){
                    count+=1;
                }
                ans = max(ans, count);

                if (v.count(s[i])){
                    count-=1;
                }

                i+=1;
            }
            j+=1;
        }

        return ans;
    }
};

File: 1567-maximum-number-of-vowels-in-a-substring-of-given-length/maximum-number-of-vowels-in-a-substring-of-given-length.go
var vowels = map[byte]bool {
    'a' : true,
    'e' : true,
    'i' : true,
    'o' : true,
    'u' : true,
}

func maxVowels(s string, k int) int {
    cnt := 0
    max_cnt := 0
    i, j := 0, 0
    for j < len(s) {
        char := s[j]
        if _, ok := vowels[char]; ok {
            cnt++
        }

        for (j - i + 1) > k {
            if _, ok := vowels[s[i]]; ok {
                cnt--
            }
            i++
        }

        if (j - i + 1) == k {
            max_cnt = max(max_cnt, cnt)
        }

        j++
    }
    return max_cnt
}